plugins/packages/athena/lib/index.ts

```diff
@@ -0,0 +1,66 @@
+import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-plugins/common';
+import { SourceOptions, QueryOptions } from './types';
+const AthenaExpress = require('athena-express');
+const AWS = require('aws-sdk');
+
+export default class Athena implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions): Promise<QueryResult> {
+    let result = {};
+    const athenaClient = await this.getConnection(sourceOptions);
+    const myQuery = {
+      sql: queryOptions.query,
+      db: sourceOptions.database,
+      ...(queryOptions.pagination?.length > 0 && { pagination: queryOptions.pagination }),
+      ...(queryOptions?.nextToken?.length > 0 && { nextToken: queryOptions.nextToken }),
+      ...(queryOptions?.queryExecutionId?.length > 0 && { queryExecutionId: queryOptions.queryExecutionId }),
+    };
+
+    try {
+      const data = await athenaClient.query(myQuery);
+      result = this.toJson(data);
+    } catch (error) {
+      throw new QueryError('Query could not be completed', error.message, {});
+    }
+
+    return {
+      status: 'ok',
+      data: result,
+    };
+  }
+
+  async testConnection(sourceOptions: SourceOptions): Promise<ConnectionTestResult> {
+    const athenaClient = await this.getConnection(sourceOptions);
+    try {
+      await athenaClient.query('SHOW TABLES');
```

Comment:
why are we using try catch here? it doesn't make any difference since we are throwing the same error